vyos/vyos-1x
# sshguard: T4408: Add service ssh dynamic-protection

## Change Summary

Sshguard protects hosts from brute-force attacks
Can inspect logs and block "bad" addresses by threshold
Auto-generate rules for nftables
When service stopped all generated rules are deleted
nft "type filter hook input priority filter - 10"

Requires pkg `sshguard` from vyos-build

## Related PR
https://github.com/vyos/vyos-build/pull/233

## Types of changes


## Related Task(s)

* https://phabricator.vyos.net/T4408

## Component(s) name

ssh, sshguard
## Proposed changes


## How to test
Minimal configuration:
```
set service ssh dynamic-protection
```
Possible options:
```
set service ssh dynamic-protection block-time 120
set service ssh dynamic-protection detect-time 1800
set service ssh dynamic-protection threshold 30
set service ssh dynamic-protection whitelist-address 192.0.2.1
```

Check ruleset:
```
table ip sshguard {
	set attackers {
		type ipv4_addr
		flags interval
	}

	chain blacklist {
		type filter hook input priority filter - 10; policy accept;
		ip saddr @attackers drop
	}
}
table ip6 sshguard {
	set attackers {
		type ipv6_addr
		flags interval
	}

	chain blacklist {
		type filter hook input priority filter - 10; policy accept;
		ip6 saddr @attackers drop
	}
}
```
Check sshguard.config:
```
vyos@tstrtr2# cat /etc/sshguard/sshguard.conf | egrep -v "#"

BACKEND="/usr/libexec/sshguard/sshg-fw-nft-sets"

LOGREADER="LANG=C journalctl -afb -p info -n1 -t sshd -o cat"

THRESHOLD=30

BLOCK_TIME=120

DETECTION_TIME=1800

WHITELIST_FILE=/etc/sshguard/whitelist
```

## Checklist:
